master/buildbot/status/web/jsontestresults.py

```diff
@@ -50,48 +50,58 @@ def content(self, req, cxt):
         cxt['splitext'] = splitext
         cxt['selectedproject'] = project
         cxt['removeTestFilter'] = removeTestFilter
+        cxt['results'] = {
+            0: 'Inconclusive',
+            1: 'NotRunnable',
+            2: 'Skipped',
+            3: 'Ignored',
+            4: 'Success',
+            5: 'Failed',
+            6: 'Error',
+            7: 'Cancelled'
+        }
+
+        json_data = None
+        error_message = None
 
         try:
-
             json_data = json.loads(self.log.getText())
 
-            if json_data is None:
-                raise ValueError("Json object is None")
-
-            cxt['data'] = json_data
-
-            if json_data['summary']:
-                success_count = json_data['summary']['successCount']
-                total_count = json_data['summary']['testsCount']
-                if success_count != 0 and total_count != 0:
-                    success_per = (float(success_count) / float(total_count)) * 100.0
-                    json_data['summary']['success_rate'] = success_per
-
-            json_data['filters'] = {
-                'Inconclusive': True,
-                'Skipped': False,
-                'Ignored': False,
-                'Success': False,
-                'Failed': True,
-                'Error': True,
-                'Cancelled': True
-            }
-
-            cxt['results'] = {
-                0: 'Inconclusive',
-                1: 'NotRunnable',
-                2: 'Skipped',
-                3: 'Ignored',
-                4: 'Success',
-                5: 'Failed',
-                6: 'Error',
-                7: 'Cancelled'
-            }
-
-        except ValueError as e:
-            log.msg("Error with parsing json: {0}".format(e))
+            if json_data is not None:
+                if 'summary' in json_data:
+                    success_count = json_data['summary']['successCount']
+                    total_count = json_data['summary']['testsCount']
+                    if success_count != 0 and total_count != 0:
+                        success_per = (float(success_count) / float(total_count)) * 100.0
+                        json_data['summary']['success_rate'] = success_per
+
+                json_data['filters'] = {
+                    'Inconclusive': True,
+                    'Skipped': False,
+                    'Ignored': False,
+                    'Success': False,
+                    'Failed': True,
+                    'Error': True,
+                    'Cancelled': True
+                }
+
+                cxt['data'] = json_data
+            else:
+                error_message = "[{0}] Error occurred while parsing JSON test report data" \
+                    .format(self.__class__.__name__)
         except KeyError as e:
-            log.msg("Key error in json: {0}".format(e))
+            error_message = "[{0}] Key error in json: {1}".format(self.__class__.__name__, e)
+        except:
```

Comment:
@kkateq also could you check this one to except Exception: ? and add a message to the log with the failure? 

```
            except Exception:
                log.msg(Failure(), "Unexpected exception caught while loading JSON test report data")
```